scripts/image-names.sh

```diff
@@ -0,0 +1,185 @@
+#!/bin/bash
+
+# This is a quick and dirty script to generate image_vars.src that is compatible with Bash 3.
+# This script can be replaced with for example a Gradle task later.
+
+# TODO: we should be more consistent about naming our images. This script will
+#       help us migrate to get consistency
+
+gradle_folder() {
+  case "$1" in
+    pdfgenerator|assessclaim*) echo "./service-python/$1";;
+    *) echo "./$1";;
+  esac
+}
+
+gradle_image_name() {
+  case "$1" in
+    svc-lighthouse-api) echo "va/abd_vro-service-data-access";; # TODO: update image name
+    *) echo "va/abd_vro-$1";;
+  esac
+}
+
+helm_image_key() {
+  case "$1" in
+    postgres) echo "db";;
+    db-init) echo "dbInit";;
+    svc-lighthouse-api) echo "serviceDataAccess";;       # TODO: rename to svcLighthouseApi
+    pdfgenerator) echo "pdfGenerator";;                  # TODO: rename to svcPdfGenerator
+    assessclaimdc7101) echo "serviceAssessClaimDC7101";; # TODO: rename to svcAssessorDc7101
+    assessclaimdc6602) echo "serviceAssessClaimDC6602";; # TODO: rename to svcAssessorDc6602
+    app|*) echo "$1";;
+  esac
+}
+
+bash_var_prefix() {
+  helm_image_key "$@"
+}
+
+nonprod_image_name() {
+  VROENV=${2:-dev}
+  echo "${VROENV}_`secrel_image_name $1`"
+}
+
+secrel_image_name() {
+  case "$1" in
+    svc-lighthouse-api) echo "vro-service-data-access";; # TODO: update image name
+    pdfgenerator|assessclaim*) echo "vro-service-$1";;
+    *) echo "vro-$1";;
+  esac
+}
+
+secrel_dockerfile() {
+  case "$1" in
+    *) echo "`gradle_folder $1`/src/docker/Dockerfile";;
+  esac
+}
+
+secrel_docker_context() {
+  case "$1" in
+    pdfgenerator|assessclaim*) echo "`gradle_folder $1`/src";;
+    app|*) echo "`gradle_folder $1`/src/main/resources";;
+  esac
+}
+
+# These names should match directory names
+IMAGES=( app postgres db-init svc-lighthouse-api pdfgenerator assessclaimdc7101 assessclaimdc6602)
+echo
+echo "=== ${#IMAGES[@]} VRO images"
+for INDEX in ${!IMAGES[@]}; do
+  echo "[$INDEX]: ${IMAGES[$INDEX]}"
+done
+echo
+echo "=== Verifying folders and files"
+{
+for IMG in ${IMAGES[@]}; do
+  echo "--- $IMG"
+  ls `gradle_folder $IMG`/build.gradle
+  ls `secrel_dockerfile $IMG`
+  echo
+done
+} >> /dev/null
+
+SRC_FILE=scripts/image_vars.src
+echo
+echo "=== Overwriting $SRC_FILE"
+{
+  VAR_PREFIXES=()
+  for IMG in ${IMAGES[@]}; do
+    VAR_PREFIXES+=(`bash_var_prefix $IMG`)
+  done
+
+echo "# This file is autogenerated -- update and re-run $0 instead."
+echo "# Usage: source \"$SRC_FILE\""
+echo '# These variables are used in GitHub Actions and deploy scripts.
+
+# Array of VRO images'
+echo "VRO_IMAGES_ARR=( ${IMAGES[@]} )"
+echo '# Usage: for IMG in ${VRO_IMAGES_ARR[@]}; do echo "- $IMG"; done'
+echo "export VRO_IMAGES=\"${IMAGES[@]}\""
+echo
+echo '# Array of variable prefixes'
+echo "VAR_PREFIXES_ARR=( ${VAR_PREFIXES[@]} )"
```

Comment:
Did you mean to export this? BTW I am not sure why you named this ".src". If you name .sh it will be checked by shellcheck. In addition to this I see a one more warning I think. Or maybe add this to shellcheck list.

Reply:
Note https://github.com/department-of-veterans-affairs/abd-vro/pull/535/files#diff-998d07e8179e6225d10250a88d236a568ae5946f558fd9e7bb9fd1c2e600e1edR21:
```
# Note: Bash arrays cannot be exported; use this workaround to
#       set the array variable from the string first:
```

Reply:
I use `.src` to mean the file is intended to be `source`d rather than just run like other `.sh` files.

Reply:
Running shellcheck on the src file manually shows irrelevant errors:
```
❯ shellcheck scripts/image_vars.src

In scripts/image_vars.src line 7:
VRO_IMAGES_ARR=( app postgres db-init svc-lighthouse-api pdfgenerator assessclaimdc7101 assessclaimdc6602 )
^------------^ SC2034 (warning): VRO_IMAGES_ARR appears unused. Verify use (or export if used externally).


In scripts/image_vars.src line 12:
VAR_PREFIXES_ARR=( app db dbInit serviceDataAccess pdfGenerator serviceAssessClaimDC7101 serviceAssessClaimDC6602 )
^--------------^ SC2034 (warning): VAR_PREFIXES_ARR appears unused. Verify use (or export if used externally).

For more information:
  https://www.shellcheck.net/wiki/SC2034 -- VAR_PREFIXES_ARR appears unused. ...
```

Reply:
[add 'shellcheck disable' in src file](https://github.com/department-of-veterans-affairs/abd-vro/pull/535/commits/a025fad87b07d0715f64c2582b2e4ccb4414773c)
```
❯ shellcheck scripts/image_vars.src

```
No alerts

Reply:
@au2 `build.gradle` updated to scan `.src` files.